### PR TITLE
Starting v2 release following API break

### PR DESCRIPTION
**Pull request by dims**
In cd60aa438f9c750641d5feacd2999e58df8af28f, we changed the API leading
to issues like:
```
src/k8s.io/client-go/transport/round_trippers.go:70:11: cannot convert klog.V(9) (type klog.Verbose) to type bool
```

So we should move to v2 of klog

Change-Id: Ibf270e0e6dd326fa91d18140e92139455cbe4de0

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by DirectXMan12** [comment]
I replied a bit to your comment in https://github.com/kubernetes/klog/pull/20#issuecomment-523140608 and https://github.com/kubernetes/klog/pull/20#issuecomment-523141779.

TL;DR: this is a decent stopgap till we can get through a KEP to switch all of client-go and friends over to structured logging natively.  That's a lot more work longterm, though (I'd like to get it done, but even if we were to start now, it'd still be a while).

**Comment by dims** [comment]
+1 to get this in AND cut a release-1.x branch AND v1.0.0 tag at the commit just prior to #20

**Comment by liggitt** [comment]
I would recommend reverting #20, changing to v2, then considering reintroducing #20

**Comment by DirectXMan12** [comment]
we may want to do the /v2 branch approach as well, since a) it gives us a path to do bugfixes on v1 and b) considering the number of people trying to `go get` w/o module support.

**Comment by liggitt** [comment]
do you need to change `go_import_path: k8s.io/klog/v2` in travis?

**Comment by liggitt** [comment]
and update the readme to note the minimum go version

**Comment by liggitt** [comment]
might need something like this in travis:
```
go_import_path: k8s.io/klog/v2
dist: xenial
go:
  - 1.11.x
  - 1.12.x
  - 1.13.x
```

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,7 @@
 language: go
-go_import_path: k8s.io/klog
+go_import_path: k8s.io/klog/v2
 dist: xenial
 go:
-  - 1.9.x
-  - 1.10.x
   - 1.11.x
   - 1.12.x
```
Comment (liggitt) [inline .travis.yml]: add 1.13.x?

**Comment by liggitt** [comment]
```
=== RUN   TestHeaderWithDir
--- FAIL: TestHeaderWithDir (0.00s)
    klog_test.go:217: log format error: 0 elements, error input does not match format:
        I0102 15:04:05.067890    1234 v2/klog_test.go:212] test
    klog_test.go:223: log format error: got:
        	"I0102 15:04:05.067890    1234 v2/klog_test.go:212] test\n"
        want:	"I0102 15:04:05.067890    1234 klog/klog_test.go:0] test\n"
```

File: .travis.yml (hunk 1 above)
Comment (dims) [reply .travis.yml]: done

**Comment by dims** [comment]
fixing test
